src/web/components/ClientSideTokenGeneration/CstgDomainsTable.tsx

```diff
@@ -21,10 +23,50 @@ export function CstgDomainsTable({
   onUpdateDomains,
   onAddDomains,
 }: CstgDomainsTableProps) {
+  const initialRowsPerPage = 10;
+  const initialPageNumber = 1;
+
   const [showAddDomainsDialog, setShowAddDomainsDialog] = useState<boolean>(false);
   const [showDeleteDomainsDialog, setShowDeleteDomainsDialog] = useState<boolean>(false);
   const [selectedDomains, setSelectedDomains] = useState<string[]>([]);
+  const [searchedDomains, setSearchedDomains] = useState<string[]>(domains);
+  const [pagedDomains, setPagedDomains] = useState<string[]>(domains);
+  const [searchText, setSearchText] = useState('');
+
+  const [pageNumber, setPageNumber] = useState<number>(initialPageNumber);
+  const [rowsPerPage, setRowsPerPage] = useState<RowsPerPageValues>(initialRowsPerPage);
+
   const isSelectedAll = domains.length && domains.every((d) => selectedDomains.includes(d));
+
+  useEffect(() => {
+    if (pageNumber && rowsPerPage) {
+      if (searchText === '') {
+        setSearchedDomains(sortStrings(domains));
+      }
+      setPagedDomains(
+        sortStrings(searchedDomains).filter(
+          (_, index) =>
+            index >= (pageNumber - 1) * rowsPerPage &&
+            index < (pageNumber - 1) * rowsPerPage + rowsPerPage
+        )
+      );
+    } else {
+      setSearchedDomains(sortStrings(domains));
+      setPagedDomains(
+        sortStrings(domains).filter(
+          (_, index) => index >= initialPageNumber - 1 && index < initialRowsPerPage
+        )
+      );
+    }
+  }, [domains, pageNumber, rowsPerPage, searchText, searchedDomains]);
+
+  useEffect(() => {
+    if (searchText !== '') {
+      setPageNumber(initialPageNumber);
+      setRowsPerPage(initialRowsPerPage);
+    }
+  }, [searchedDomains.length, searchText]);
```

Comment:
When the search changes, I found it best to just go back to the first page and initial rows per page for both functionality and UX